packages/babel-plugin-transform-async-to-generator/test/fixtures/async-to-generator/async-iife-with-regenerator-spec/expected.js

```diff
@@ -0,0 +1,74 @@
+var _this = this;
+
+babelHelpers.asyncToGenerator(
+/*#__PURE__*/
+regeneratorRuntime.mark(function _callee() {
+  return regeneratorRuntime.wrap(function _callee$(_context) {
+    while (1) switch (_context.prev = _context.next) {
+      case 0:
+        _context.next = 2;
+        return 'ok';
+
+      case 2:
+      case "end":
+        return _context.stop();
+    }
+  }, _callee, this);
+}))();
+babelHelpers.asyncToGenerator(
+/*#__PURE__*/
+regeneratorRuntime.mark(function _callee2() {
+  return regeneratorRuntime.wrap(function _callee2$(_context2) {
+    while (1) switch (_context2.prev = _context2.next) {
+      case 0:
+        babelHelpers.newArrowCheck(this, _this);
+        _context2.next = 3;
+        return 'ok';
+
+      case 3:
+      case "end":
+        return _context2.stop();
+    }
+  }, _callee2, this);
+})).bind(this)();
+
+/*#__PURE__*/
+(function () {
+  var _notIIFE = babelHelpers.asyncToGenerator(
+  /*#__PURE__*/
+  regeneratorRuntime.mark(function _callee3() {
+    return regeneratorRuntime.wrap(function _callee3$(_context3) {
+      while (1) switch (_context3.prev = _context3.next) {
+        case 0:
+          _context3.next = 2;
+          return 'ok';
+
+        case 2:
+        case "end":
+          return _context3.stop();
+      }
+    }, _callee3, this);
+  }));
+
+  return function notIIFE() {
+    return _notIIFE.apply(this, arguments);
+  };
+})();
+
+/*#__PURE__*/
+babelHelpers.asyncToGenerator(
+/*#__PURE__*/
+regeneratorRuntime.mark(function _callee4() {
+  return regeneratorRuntime.wrap(function _callee4$(_context4) {
+    while (1) switch (_context4.prev = _context4.next) {
+      case 0:
+        babelHelpers.newArrowCheck(this, _this);
+        _context4.next = 3;
+        return 'not iife';
+
+      case 3:
+      case "end":
+        return _context4.stop();
+    }
+  }, _callee4, this);
+})).bind(this);
```

Comment:
The call to `.bind(this)` makes this not eligible for DCE by Uglify, even with the `#__PURE__` annotation.

Maybe changing this to just check for `MemberExpression` at all is sufficient; or should we stay optimistic and add the `#__PURE__` annotation in this case? 🤔

Reply:
Is `.bind` impure only when the user modifies `Function.prototype.bind`? If yes, I vote for being optimistic (since Babel has always assumed that builtins are unchanged), otherwise we should remove it.